### PR TITLE
created contributing.md

### PR DESCRIPTION
**Pull request by Sakshamgupta90**
Fix Issue #103

### CODE REVIEW

**Comment by Sakshamgupta90** [comment]
Hi @ocefpaf, I couldn't able to understand why this pre-commit check gets failed, can you clear this out?

**Comment by ocefpaf** [comment]
> Hi @ocefpaf, I couldn't able to understand why this pre-commit check gets failed, can you clear this out?

If you click on the link it will list all the failures. They ar mostly related to long lines and we can fix them in another PR. The file you added seems to be OK.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
+# Contributors Guide
+
+Interested in helping out the IOOS QARTOD?
```
Comment (ocefpaf) [inline CONTRIBUTING.md:3]: QARTOD is 1 module of `ioos_qc`. There are others QA/QC standards implemented here.

```suggestion
Interested in helping out the ioos_qc?
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+Have a few minutes to tackle an issue? Or improve the documentation?
+In this guide we will get you setup and integrated into contributing to IOOS QARTOD!
+
+## Introduction
+
+First off, thank you for considering contributing to IOOS QARTOD!
```
Comment (ocefpaf) [inline CONTRIBUTING.md:9]:
```suggestion
First off, thank you for considering contributing to ioos_qc!
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+QARTOD is collection of utilities, scripts and tests to assist in automated quality assurance and quality control for oceanographic datasets and observing systems.
```
Comment (ocefpaf) [inline CONTRIBUTING.md:11]: Please replace all QARTOD references to the name to the library ` ioos_qc`.

**Comment by Sakshamgupta90** [comment]
So, is the documentation written is fine enough?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+We recommend using the [conda](https://conda.io/docs/) package manager for your Python environments.
+
+Install [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+(link with instructions) on your system if not already available
+(check with ``git --version`` at the command line.)
+This can also be installed from a variety of package managers, including ``conda`` if needed.
+
+Login to your [GitHub](https://github.com) account and make a fork of the
+[IOOS_QC repository](https://github.com/ioos/ioos_qc) by clicking the "Fork" button.
```
Comment (ocefpaf) [inline CONTRIBUTING.md:64]: It is odd but the library name should be lower case.

```suggestion
[ioos_qc repository](https://github.com/ioos/ioos_qc) by clicking the "Fork" button.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+```sh
+git remote add upstream https://github.com/ioos/ioos_qc.git
+```
+
+Create a new conda environment for us to configure, and give it a name.
+
+For Environment Setup:
+
+After ``-n`` you can specify any name you'd like; here we've chosen ``ioosqc38``.
```
Comment (ocefpaf) [inline CONTRIBUTING.md:90]:
```suggestion
After ``-n`` you can specify any name you'd like; here we've chosen ``ioosqc312`.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+```sh
+conda create -n ioosqc38 python=3.8
```
Comment (ocefpaf) [inline CONTRIBUTING.md:93]: Let's start with latest.

```suggestion
conda create -n ioosqc38 python=3.12
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+```
+
+**IMPORTANT**: Always activate this environment when developing and testing your changes!
+
+```sh
+conda activate ioosqc38
```
Comment (ocefpaf) [inline CONTRIBUTING.md:99]:
```suggestion
conda activate ioosqc312
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+conda install --file requirements.txt --file requirements-dev.txt
+```
+
+Note that those requirement file names may change depending on the repository.
+Again, always check the local README for more information.
+
+Finally, to run all the tests!
+
+```sh
+pytest
```
Comment (ocefpaf) [inline CONTRIBUTING.md:115]: Please check the line we have in the GitHub Actions file and use it here. We have some extra options.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+```
+
+Once you are done with environemtn setup, its time to build docs:
+
+Create the following environment:
+
+```sh
+conda create -y -n ioosqc38_docs python=3.8
```
Comment (ocefpaf) [inline CONTRIBUTING.md:123]: Please change all references of Python 3.8 to Python 3.12

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+
+```sh
+conda activate ioosqc38_docs
+```
+
+You will have to do this any time you re-open your prompt.
+Currently there are no packages in this environment, let's change that.
+
+```sh
+conda install -y --file requirements.txt \ --file requirements-dev.txt \ --file docs/requirements.txt
```
Comment (ocefpaf) [inline CONTRIBUTING.md:136]: We should simplify this later but it is OK for now. We don't need so many requirements file. We should probably add those to pyproject.toml.

```suggestion
conda install  --file requirements.txt --file requirements-dev.txt --file docs/requirements.txt
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+
+Note that those requirement file names may change depending on the repository.
+Again, always check the local README for more information.
+
+```sh
+cd docs
+make html
+```
+Then open a browser to:
+``build/html/index.html``
```
Comment (ocefpaf) [inline CONTRIBUTING.md:147]: Did you try building our docs? Are you familiar with sphinx? That could be a nice exercise to understand a bit more of our infrastructure.

File: CONTRIBUTING.md (hunk 5 above)
Comment (Sakshamgupta90) [reply CONTRIBUTING.md:90]: But in the documentation its, written the ioosqc38, so I guess we need to fix that as well.

Comment (ocefpaf) [reply CONTRIBUTING.md:90]: The contributing guide will replace any contributing instructions in the docs. We will remove those later in another PR. That is why I requested you to try to build it in a comment below.

File: CONTRIBUTING.md (hunk 11 above)
Comment (Sakshamgupta90) [reply CONTRIBUTING.md:147]: Yes, I tried building the docs. And no, I am not familiar with sphinx.

Comment (ocefpaf) [reply CONTRIBUTING.md:147]: Did you succeed? If not what were to problems you faced that could be added as instructions here?

Comment (Sakshamgupta90) [reply CONTRIBUTING.md:147]: Yes, I got succeeded, do received some warnings, but it build successfully.

Comment (ocefpaf) [reply CONTRIBUTING.md:147]: Interesting. Do you want to tackle those warnings? Any missing packages that you had to install that were not in the requirement.txt file?

Comment (Sakshamgupta90) [reply CONTRIBUTING.md:147]: No packages needed. And it will build automatically, nothing to tackle with.

Comment (ocefpaf) [reply CONTRIBUTING.md:147]: Interesting. I had to install `pooch`, which is not specified in any of our requirement files, and adapt an old notebook with a new syntax for `erddapy`. I wonder if your build ran the notebooks correctly then, you should've seen these failures.

Anyways, let's tackle the build warnings in a next PRs.

Comment (Sakshamgupta90) [reply CONTRIBUTING.md:147]: Oh my bad!
I just tried to run the notebooks on jupyter, and there it shows those module errors. Sorry for that, I thought once the build is done, everything is fine.
So, I will add those missing modules in the requirements files. Will open the PR for its same.
So sorry for that once again.

**Pull request merged**